### ISSUE TITLE
Rstudio does respect environment variables initiated during login shell.

### ISSUE THREAD

**Issue by d-walkama** [jupyterhub/jupyter-rsession-proxy#135]
We spawn single user jupyterhub instances by setting the c.Spawner.cmd to a script that initiates a login shell with "!#/bin/bash -l". This allows us to dereference shell variables that were created in .Renviron and we see this works by checking them in R console within JupyterHub. However, this does not work in Rstudio using jupyter-rsession-proxy. Is there any way to solve this issue?

**Comment by ryanlovett** [comment]
I tried setting variables in ~/.Renviron in different spawning environments (kubespawner, batchspawner) on two separate hubs and both were available within the RStudio console. As far as I know, jupyter-rsession-proxy cannot alter this behavior. However can you elaborate on 

> dereference shell variables that were created in .Renviron

That sounds like you're doing something different than vanilla `Key1=value1` assignments.

**Comment by d-walkama** [comment]
Yes we wish to do something a bit different than hardcoding the variables. i.e setting:

PATH=${PATH}

in .Renviron, so that we can build the variables during the login (scripts in /etc/profile.d) and pull them into R. We've been able to do this just fine in the R console within JupyterHub, but not with Rstudio.

**Comment by ryanlovett** [comment]
Just trying to follow this example, you define things like PATH in /etc/profile.d/, but if you do not set PATH in .Renviron, RStudio won't inherit it from the shell environment?

Do you know if this also works with RStudio outside of jupyterhub + jupyter-rsession-proxy ?

**Comment by d-walkama** [comment]
Correct, I believe R only creates variables that are set in .Renviron. I.e. if you have a shell variable TEMP="test" and you do not set TEMP=${TEMP} in .Renviron, then Sys.getenv("TEMP") will return a null value.

**Comment by d-walkama** [comment]
I see this spot in your code which I'm assuming is where the environment is stripped down to just the specific things listed below?:

`def _get_env(port):
        # Detect various environment variables rsession requires to run
        # Via rstudio's src/cpp/core/r_util/REnvironmentPosix.cpp
        cmd = ['R', '--slave', '--vanilla', '-e',
                'cat(paste(R.home("home"),R.home("share"),R.home("include"),R.home("doc"),getRversion(),sep=":"))']

        r_output = subprocess.check_output(cmd)
        R_HOME, R_SHARE_DIR, R_INCLUDE_DIR, R_DOC_DIR, version = \
            r_output.decode().split(':')

        return {
            'R_DOC_DIR': R_DOC_DIR,
            'R_HOME': R_HOME,
            'R_INCLUDE_DIR': R_INCLUDE_DIR,
            'R_SHARE_DIR': R_SHARE_DIR,
            'RSTUDIO_DEFAULT_R_VERSION_HOME': R_HOME,
            'RSTUDIO_DEFAULT_R_VERSION': version,
        }
`

**Comment by ryanlovett** [comment]
That code does set some environment variables, but it should not unset everything else. jupyter-server-proxy [integrates those variables](https://github.com/jupyterhub/jupyter-server-proxy/blob/main/jupyter_server_proxy/handlers.py#L645) into the existing environment.

I have a feeling this has more to due with the spawner or RStudio and not jupyter-rsession-proxy. However I'm not sure why RStudio is not otherwise picking up your /etc/profile.d/ configuration. According to RStudio's docs, R is [started under a bash login shell](https://docs.rstudio.com/ide/server-pro/1.0.34/r-sessions.html#customizing-session-launches) and should read /etc/profile.

**Comment by d-walkama** [comment]
I think there are major differences between how the open-source Rstudio server and Rstudio server pro configure themselves.

**Comment by scj643** [comment]
You could shim the rsession executable with a shell script and use that to inject environment variables. Using `--rsession-path=` in the rstudio command.
